test/iaas.GcpClient.spec.js

```diff
@@ -0,0 +1,538 @@
+'use strict';
+const _ = require('lodash');
+const GcpStorage = require('@google-cloud/storage');
+const GcpCompute = require('@google-cloud/compute');
+const Promise = require('bluebird');
+const logger = require('../lib/logger');
+const errors = require('../lib/errors');
+const utils = require('../lib/utils');
+const lib = require('../lib');
+const GcpClient = lib.iaas.GcpClient;
+const NotFound = errors.NotFound;
+const Forbidden = errors.Forbidden;
+
+const CONNECTION_WAIT_SIMULATED_DELAY = 5;
+const config = {
+  backup: {
+    retention_period_in_days: 14,
+    max_num_on_demand_backup: 2,
+    status_check_every: 120000, // (ms) Check the status of backup once every 2 mins
+    backup_restore_status_poller_timeout: 86400000, // (ms) Deployment backup/restore must finish within this timeout time (24 hrs)
+    backup_restore_status_check_every: 120000, // (ms) Check the status of deployment backup/restore once every 2 mins
+    abort_time_out: 300000, //(ms) Timeout time for abort of backup to complete
+    provider: {
+      name: 'gcp',
+      projectId: 'my-gcp-dev',
+      credentials: {
+        'type': 'service_account',
+        'project_id': 'my-gcp-dev',
+        'private_key_id': 'kkkkkkkk',
+        'private_key': '-----BEGIN PRIVATE KEY-----\nlllllllllllllllllllllll\n-----END PRIVATE KEY-----\n',
+        'client_email': 'my-test-user@my-gcp-dev.iam.myaccounts.com',
+        'client_id': '000000000000000',
+        'auth_uri': 'https://myaccounts.com/oauth2/auth',
+        'token_uri': 'https://myaccounts.com/oauth2/token',
+        'auth_provider_x509_cert_url': 'https://myaccounts.com/oauth2/v1/certs',
+        'client_x509_cert_url': 'https://myaccounts.com/v1/metadata/x509/my-test-user%40my-gcp-dev.iam.myaccounts.com'
+      },
+      container: 'sample-container',
+      provider: 'gcp'
+    }
+  }
+};
+const settings = config.backup.provider;
+const storageUrl = 'https://myaccounts.com/storage/v1';
+const bucketUrl = '/b';
+const region = 'EUROPE-WEST1';
+const storageClass = 'REGIONAL';
+const bucketMetadataResponse = [{},
+  {
+    kind: 'storage#bucket',
+    id: 'sample-container',
+    selfLink: 'https://myaccounts.com/storage/v1/b/sample-container',
+    projectNumber: '00000000000',
+    name: 'sample-container',
+    timeCreated: '2017-12-24T10:23:50.348Z',
+    updated: '2017-12-24T10:23:50.348Z',
+    metageneration: '1',
+    location: 'EUROPE-WEST1',
+    storageClass: 'REGIONAL',
+    etag: 'aaa'
+  }
+];
+const getFilesResponse = [
```

Comment:
is this supposed to be an array of array?

Reply:
Yes, getFiles call returns array in which 0th element is again array of file. Please see [Bucket#getFiles](https://cloud.google.com/nodejs/docs/reference/storage/1.6.x/Bucket#getFiles) and [GetFilesResponse](https://cloud.google.com/nodejs/docs/reference/storage/1.6.x/global#GetFilesResponse)